### PR TITLE
Don't get items in __init__

### LINKED ISSUE #4: Glossary not listing private Terms to users with permissions to view them
imagem da pagina 2015-05-15 - 14 09 42

### PR DESCRIPTION
**Pull request by jkubaile**
This fixes #4 

Problem was, that making catalog searches in the browser view's init method doesnt work cause of missing acquisition chain (see http://docs.plone.org/develop/plone/views/browserviews.html#view-init-method-special-cases)

Cause "prepare_item{s}" as name doesnt make sense anymore, i renamed them to get_items.

I fixed the unit tests, but had problem running the robot ones. So I cant look at them.

### CODE REVIEW

**Comment by jkubaile** [comment]
Mhh, I just added german translations in my forked branch and it gets also part of this pull request. Sorry, didnt know this. But feel free to add them, too ;)

**Comment by rodfersou** [comment]
@jkubaile please add a test to simulate the problem

**Comment by hvelarde** [comment]
@jkubaile please push the German translations direct on master branch so we can focus only on the issue here.

I agree `prepare_item` is not a good name but I would prefer to use `get_glossary_entries` or `get_entries` instead.

as @rodfersou said, add a test to demonstrate the issue first; also, those changes added to comply with PEP 8 are needless, IMO, and take our attention out of the issue.

**Comment by hvelarde** [comment]
@jkubaile please rebase your fork so CI tests get fixed.

**Comment by jkubaile** [comment]
Ok, thanks for the feedback.

I pushed the translation stuff upstream.

I also renamed the method to get_entries.

But writing tests for this case is a bit complicated in my opinion. I have no workflows in the current setup to reproduce the stuff in a test case - also i am not sure if it will trigger via api.content.get_view(). On the other hand the problem is well known and documented in the link I already provided. Maybe we should not make it more complicated as it is ;)

File: src/collective/glossary/browser/views.py [hunk 1]
```diff
@@ -23,10 +24,9 @@ class TermView(BrowserView):
     def __init__(self, context, request):
         self.context = context
         self.request = request
-        self.item = self.prepare_item()
 
-    def prepare_item(self):
-        """Prepare term in the desired format"""
+    def get_entry(self):
+        """get term in the desired format"""
```
Comment (hvelarde) [inline src/collective/glossary/browser/views.py]: *Get* instead of *get*

File: src/collective/glossary/browser/views.py [hunk 2]
```diff
@@ -135,11 +136,11 @@ class JsonView(BrowserView):
...
         self.context = context
         self.request = request
-        self.items = self.prepare_items()
 
     @ram.cache(_catalog_counter_cachekey)
-    def prepare_items(self):
-        """Get all itens and prepare in the desired format"""
+    def get_json_entries(self):
+        """Get all itens and prepare in the desired format.
+        Note: do not name it get_entries, otherwise caching is broken. """
```
Comment (hvelarde) [inline src/collective/glossary/browser/views.py]: why?

File: src/collective/glossary/tests/test_views.py [hunk 3]
```diff
@@ -149,11 +156,22 @@ class JsonViewTestCase(BaseViewTestCase):
 
     def setUp(self):
         super(JsonViewTestCase, self).setUp()
-        self.view = api.content.get_view(u'glossary', self.portal, self.request)
+        self.view = api.content.get_view(u'glossary',
```
Comment (hvelarde) [inline src/collective/glossary/tests/test_views.py]: I would prefer something like this to follow plone.api conventions:

```python
self.view = api.content.get_view(
    name=u'glossary',
    context=self.portal,
    request=self.request
)
```

File: src/collective/glossary/tests/test_views.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 from zope.publisher.browser import TestRequest
 
 import unittest
+import json
```
Comment (hvelarde) [inline src/collective/glossary/tests/test_views.py]: do you mind moving this import to the only test using the json module?

File: src/collective/glossary/browser/views.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 from collective.glossary.interfaces import ITerm
 from plone import api
 from plone.memoize import ram
+
```
Comment (hvelarde) [inline src/collective/glossary/browser/views.py]: remove this empty line

**Comment by hvelarde** [comment]
@jkubaile is easier to open a branch instead of a fork so we can help on the fixing of small stupid things.

File: src/collective/glossary/browser/views.py (hunk 2 above)
Comment (jkubaile) [inline src/collective/glossary/browser/views.py]: Using the same function name will result in Mixing the ram cache from JsonView and GlossaryView which will not work.

**Comment by jkubaile** [comment]
@hvelarde next try ;) 

next time i will use a branch for sure!

**Comment by hvelarde** [comment]
thanks!

**Pull request merged**